lib/Graylog.js

```diff
@@ -0,0 +1,427 @@
+var zlib = require('zlib');
+var crypto = require('crypto');
+var dgram = require('dgram');
+var util = require('util');
+var EventEmitter = require('events').EventEmitter;
+var Queue = require('./Queue');
+
+var MAX_SAFE_INT = 9007199254740991;  // Number.MAX_SAFE_INTEGER
+
+/**
+ * Graylog instances emit errors. That means you really really should listen for them,
+ * or accept uncaught exceptions (node throws if you don't listen for "error").
+ */
+
+function Graylog(config) {
+    EventEmitter.call(this);
+
+	// settings (all safe to change at runtime through these setters)
+	this.setServers(config.servers);
+	this.setHostname(config.hostname || require('os').hostname());
+	this.setFacility(config.facility || 'Node.js');
+	this.setBufferSize(config.bufferSize || this.DEFAULT_BUFFERSIZE);
+	this.setDeflate(config.deflate || 'optimal');
+
+    // state
+	this._client = null;
+    this._serverIterator = 0;
```

Comment:
Ugh, I had a ton of that during this rewrite. Not sure what's up. I'll add an editorconfig file.